src/mpi/simulation.rs

```diff
@@ -113,13 +137,47 @@ impl Simulation {
                         &Event::new_person_enters_veh(agent_id, route.vehicle_id),
                     );
 
-                    let veh = Vehicle::new(route.vehicle_id, VehicleType::Network, agent);
+                    let veh = Vehicle::new(route.vehicle_id, VehicleType::Network, agent.clone());
                     self.veh_onto_network(veh, true, now);
                 }
             }
         }
     }
 
+    fn find_route(router: &mut Option<Router>, agent: &mut Agent) {
+        if let Some(router) = router {
```

Comment:
This works properly I guess. Still I think it would make more sense to have this method to have the following signature:

```
fn find_route(router: &mut Option<Router>, from_act: &Activity, to_act: &Activity) -> Vec<u64>
```
Assuming that `query_result.path.unwrap()` gives a List of Link ids. This method could even get rid of the `router` parameter if the `router` was a member.

Then the agent could have a `push_leg` method where the list of link ids as well as the travel time is passed in.